astroquery/heasarc/core.py

```diff
@@ -177,20 +174,39 @@ def query_region_async(self, position, mission, radius,
         # Submit the request
         return self.query_async(request_payload, cache=cache)
 
-    def _fallback(self, content):
+    def _old_w3query_fallback(self, content):
+        # old w3query (such as that used in ISDC) return very strange fits, with all ints
+
+        f = fits.open(BytesIO(content))
+
+        for c in f[1].columns:
+            if c.disp is not None:
+                c.format = c.disp
+            else:
+                c.format = _AsciiColumnFormat(str(c.format).replace("I", "A"))
```

Comment:
is there another way to achieve this without using a private function?

Reply:
Looks like there is a simpler way indeed, just:

```python
 c.format = str(c.format).replace("I", "A")
```

It seems to work the same, I added a small extra test to make sure it is good.

Not sure why I did not use this approach to begin with, it must have not worked, it was some time ago.